### PR TITLE
feat: Enhance the LoggingClient function

### LINKED ISSUE #294: Add support to get current log level
Currently there is support for setting the log level, but there is no support for querying the current log level.

This is very important because in some cases it is preferable to check if Debug logging is enabled before even making the Debug log call because of computations that are performed, like for instance:

```go
if lgr.IsLevelEnabled(models.DebugLog) {
  lgr.Debug(fmt.Sprintf("Average latency: %.2f ms", computeAverageLatency()))
}
```

This is a rudimentary example, but the point is that `computeAverageLatency()` may take a long time to complete, something we would prefer to avoid calling if it is never going to be actually logged.

### LINKED ISSUE #295: Add formatted alternatives to log functions, ie. logger.Infof
This could probably be a thing to be done during the work of #246.

I propose that `fmt.Sprintf`-like alternatives be added to the logging functions, like:
```go
func (lc edgeXLogger) Tracef(format string, a ...interface{})
func (lc edgeXLogger) Debugf(format string, a ...interface{})
func (lc edgeXLogger) Infof(format string, a ...interface{})
func (lc edgeXLogger) Warnf(format string, a ...interface{})
func (lc edgeXLogger) Errorf(format string, a ...interface{})
```

For one it would simplify this:
```go
loggingClient.Info(fmt.Sprintf("Starting %s %s ", internal.CoreDataServiceKey, edgex.Version))
```
into this:
```go
loggingClient.Infof("Starting %s %s ", internal.CoreDataServiceKey, edgex.Version)
```

However even more importantly, because the `fmt.Sprintf` is being called by the SDK's code, you can delay the execution of it until **_after_** the [log level check](https://github.com/edgexfoundry/go-mod-core-contracts/blob/b139196b2d205d068909f4e736248de9c8d5224f/clients/logger/logger.go#L162-L170) has been completed. In the case of some heavy formatting calls, this could save some cycles.

```go
lgr.Debug(fmt.Sprintf("%+v", deviceProfile))  // entire deviceProfile struct will be pretty-formatted no matter what
lgr.Debugf("%+v", deviceProfile)              // will only perform the formatting if Debug logging is enabled
```

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #294, fix #295 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
- Add support to get current log level 
- Add formatted alternatives to log functions

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: clients/logger/logger.go [hunk 1]
```diff
@@ -172,6 +172,8 @@ func (lc edgeXLogger) log(logLevel string, msg string, args ...interface{}) {
 
 	if args == nil {
 		args = []interface{}{"msg", msg}
+	} else if formatted {
+		args = []interface{}{"msg", fmt.Sprintf(msg, args...)}
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply clients/logger/logger.go:176]: But the go-kit log lib only accepts  key/value pair as arguments, see https://github.com/go-kit/kit/tree/master/log#structured-logging



For example:
```
func TestLogInfof(t *testing.T) {
	lc := newClient("testService", false, "", models.InfoLog)
	lc.Infof("test %s","message")
}
```
Currently, the message is:
```
level=INFO ts=2020-12-11T17:07:12.526591Z app=testService source=logger_test.go:50 msg="test message"
```
If we remove the "msg":
```
level=INFO ts=2020-12-11T17:08:07.597245Z app=testService source=logger_test.go:50 testmessage=(MISSING)
```

[the remark this answers is not in the document]
Comment (lenny-intel) [reply clients/logger/logger.go:176]: Ohh, ok. I get it. That is the key!! THX!

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by weichou1229** [comment]
Rebased.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**